### PR TITLE
[CFL] Adjust power UPD settings

### PR DESCRIPTION
**Pull request by raghavag**
This patch adjusted power related UPD settings to BIOS
UPD values are configurable through configuration data
After these changes, CPU maximum operating frequency
increased from 3.7GHZ to 4.2GHZ on WHL.

Signed-off-by: Raghava Gudla <raghava.gudla@intel.com>

### CODE REVIEW

File: Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 1]
```diff
@@ -1712,6 +1707,32 @@ UpdateFspConfig (
   FspsUpd->FspsTestConfig.PsysPmax                        = 0x0;
   FspsUpd->FspsTestConfig.CstateLatencyControl0Irtl       = 0x4e;
 
+  PowerCfgData = (POWER_CFG_DATA *) FindConfigDataByTag (CDATA_POWER_TAG);
+  if (PowerCfgData == NULL) {
+    DEBUG ((DEBUG_ERROR, "Missing PCIE RP Cfg Data!\n"));
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c]: PCIE RP  Cfg ?

File: Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  # !BSF NAME:{8-Core Ratio Limit}
+  # !BSF TYPE:{EditNum, HEX, (0,0xFF)}
+  # !BSF HELP:{LFM to Fused, For overclocking part: LFM to 255. This 8-Core Ratio Limit Must be Less than or equal to 1-Core Ratio Limit.Range is 0 to 255 0x0:0xFF}
+  gCfgData.EightCoreRatioLimit               |      * | 0x01 | 0
+
+  # !BSF NAME:{Thermal Sensor 0 Target Width}
+  # !BSF TYPE:{Combo} OPTION:{0:x1, 1:x2, 2:x4, 3:x8, 4:x16}
+  # !BSF HELP:{DMT thermal sensor suggested representative values.}
+  gCfgData.DmiTS0TW                         |      * | 0x01 | 0x3
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc]: do we need all of these ?   do we have guidance on how to set them ?

Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc]: Removed some which BIOS is not exposing.

File: Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c (hunk 1 above)
Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c]: corrected

File: Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+##
+  # !BSF PAGES:{PWR:SIL:"POWER"}
+  # !BSF PAGE:{PWR}
+  # !BSF SUBT:{CFGHDR_TMPL:POWER_CFG_DATA:0:0}
+  # !HDR EMBED:{POWER_CFG_DATA:TAG_610:START}
+
+  # !BSF NAME:{1-Core Ratio Limit}
+  # !BSF TYPE:{EditNum, HEX, (0,0xFF)}
+  # !BSF HELP:{1-Core Ratio Limit: LFM to Fused, For overclocking part: LFM to 255. This 1-Core Ratio Limit Must be greater than or equal to 2-Core Ratio Limit, 3-Core Ratio Limit, 4-Core Ratio Limit, 5-Core Ratio Limit, 6-Core Ratio Limit, 7-Core Ratio Limit, 8-Core Ratio Limit. Range is 0 to 255}
+  gCfgData.OneCoreRatioLimit                 |      * | 0x01 | 0x2C
```
Comment (myguitar) [inline Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc]: These default values are the same for CFL-R? No delta required?

Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/CfgData/CfgData_Power.dsc]: updated upd's for all 3 platforms in new patch

**Pull request merged**